### PR TITLE
refactoring to reduce cognitive complexity

### PR DESCRIPTION
**Pull request by bridgesr**
# PR checklist

Tick what applies before you raise. Delete anything that doesn't.

## Correctness
- [x] Does what the ticket asks for
- [x] Handles the edge cases (empty, null, zero, large inputs), not just the happy path
- [x] Errors are handled, not swallowed
- [x] New logic has tests, including the failure cases
- [x] Tests assert something real, not just run the code

## Keeping it simple
- [x] Doesn't rebuild something that already exists in the codebase
- [x] No more complex or slower than it needs to be
- [x] No leftover debug logging or commented-out code

## Easy to miss (a green pipeline won't flag these)
- [x] One logical change, not a pile of unrelated stuff

### If you touched the backend
- [x] Schema changes have a migration, and the Down() doesn't drop anything it shouldn't
- [x] New app settings are wired into the fa-config templates
- [x] Secrets use Key Vault references, not plain text
- [x] New outbound HTTP clients go through the shared resilience handler (AddResiliencePolicyHandler), rather than hand-rolling retries
- [x] New calls to an external service (DDEI, Egress, NetApp) have client tests against a WireMock stub (for the serialisation, auth and error handling) and are covered by the integration tests that run against the real dev services
- [x] Ran dotnet format, since CI builds and tests but doesn't check formatting

### CODE REVIEW

File: backend/CPS.ComplexCases.FileTransfer.API/Durable/Activity/TransferFile.cs [hunk 1]
```diff
@@ -136,152 +105,284 @@ public async Task<TransferResult> Run([ActivityTrigger] TransferFilePayload payl
...
+                    startTime);
 
-                telemetryEvent.FileSizeInBytes = totalSize;
-                telemetryEvent.TransferEndTime =
-                    result.SuccessfulItem?.EndTime ?? result.SkippedItem?.EndTime ?? DateTime.UtcNow;
-                telemetryEvent.IsSuccessful = result.IsSuccess || result.IsSkipped;
-                telemetryEvent.IsMultipart = result.IsSuccess && result.SuccessfulItem!.TotalPartsCount > 1;
-                telemetryEvent.TotalPartsCount = result.IsSuccess ? result.SuccessfulItem!.TotalPartsCount : 0;
+                ApplyTransferResultTelemetry(telemetryEvent, totalSize, result);
+                TrackTransferTelemetry(telemetryEvent);
```
Comment (HasanCPS) [inline backend/CPS.ComplexCases.FileTransfer.API/Durable/Activity/TransferFile.cs]: **Double telemetry on success:** `TrackTransferTelemetry` is called inside the try block (after `ApplyTransferResultTelemetry`) and then again unconditionally in `finally`. On the success path both fire, so App Insights gets a duplicate event per transfer. Remove the `TrackTransferTelemetry(telemetryEvent)` call inside the try block. Let `finally` be the single emit point for all paths.

File: backend/CPS.ComplexCases.FileTransfer.API/Durable/Orchestration/TransferOrchestrator.cs [hunk 2]
```diff
@@ -52,326 +52,391 @@ public async Task RunOrchestrator(
...
+            await RetryTransientFailuresAsync(
+                context, input, transferEntity, cleanFiles, entityId, allResults, transferOrchestrationEvent, logger);
+
+            await DeleteSourceFilesIfMoveAsync(context, input);
+
+            await FinalizeAndLogCompletionAsync(context, input);
+
+            transferOrchestrationEvent.IsSuccessful = transferOrchestrationEvent.TotalFilesFailed == 0;
+            transferOrchestrationEvent.OrchestrationEndTime = context.CurrentUtcDateTime;
+            _telemetryClient.TrackEvent(transferOrchestrationEvent);
```
Comment (HasanCPS) [inline backend/CPS.ComplexCases.FileTransfer.API/Durable/Orchestration/TransferOrchestrator.cs]: **[Same ](https://github.com/CPS-Innovation/Large-and-Complex-Cases/pull/509/changes#r3674969467)double-telemetry pattern:** `_telemetryClient.TrackEvent(transferOrchestrationEvent)` fires at the end of the try block and again in `finally`. On success both execute. Remove the `TrackEvent` call inside the try block and let `finally` handle it. Set `IsSuccessful` before the `finally` runs (it already does), or move the success flag assignment into `finally` guarded by a check.

File: backend/CPS.ComplexCases.FileTransfer.API/Durable/Activity/TransferFile.cs [hunk 3]
```diff
@@ -136,152 +105,284 @@ public async Task<TransferResult> Run([ActivityTrigger] TransferFilePayload payl
...
         }
-        catch (FileExistsException ex)
+        finally
         {
-            LogFileConflictTelemetry(payload);
-            telemetryEvent.ErrorCode = TransferErrorCode.FileExists.ToString();
-            telemetryEvent.ErrorMessage = ex.Message;
-            return CreateFailureResult(payload.SourcePath.FullFilePath ?? payload.SourcePath.Path,
-                TransferErrorCode.FileExists, MapUserMessage(TransferErrorCode.FileExists), ex);
+            telemetryEvent.TransferEndTime = DateTime.UtcNow;
```
Comment (HasanCPS) [inline backend/CPS.ComplexCases.FileTransfer.API/Durable/Activity/TransferFile.cs]: `finally` block  `telemetryEvent.TransferEndTime = DateTime.UtcNow` unconditionally overwrites the value that `ApplyTransferResultTelemetry` set from `result.SuccessfulItem?.EndTime`. The accurate end time from the actual transfer result is lost. Guard it: `telemetryEvent.TransferEndTime = telemetryEvent.TransferEndTime == default ? DateTime.UtcNow : telemetryEvent.TransferEndTime;`

File: backend/CPS.ComplexCases.FileTransfer.API/Durable/Activity/TransferFile.cs [hunk 4]
```diff
@@ -136,152 +105,284 @@ public async Task<TransferResult> Run([ActivityTrigger] TransferFilePayload payl
...
+            {
+                Rethrow = false,
+                ErrorCode = TransferErrorCode.FileExists,
+                DiagnosticMessage = fileExists.Message,
+                LogFileConflict = true,
+                Exception = fileExists
+            };
+        }
+
+        if (ex is OperationCanceledException oce && !isCancellationRequested)
```
Comment (HasanCPS) [inline backend/CPS.ComplexCases.FileTransfer.API/Durable/Activity/TransferFile.cs:290]: The two `OperationCanceledException` branches (timeout vs real cancel) are separated by ~80 lines of other exception checks. Easy to miss the second one on a future read.  Add a brief comment above the second check: `// True cancellation (token was triggered) - rethrow to let the framework handle it.`

File: backend/CPS.ComplexCases.API/Functions/MaterialBatchRename.cs [hunk 5]
```diff
@@ -91,93 +91,140 @@ public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous,
...
+        Guid volumeUuid)
+    {
+        var arg = _ontapArgFactory.CreateMaterialRenameArg(bearerToken, volumeUuid, operation.CurrentPath, operation.NewPath);
+
+        try
+        {
+            var result = await _ontapHttpClient.RenameMaterialAsync(arg);
+            return MapRenameResultToItemResult(operation, result, _logger);
+        }
+        catch (Exception ex) when (IsAuthException(ex))
```
Comment (HasanCPS) [inline backend/CPS.ComplexCases.API/Functions/MaterialBatchRename.cs:134]: `RenameSingleOperationAsync` inner `catch (Exception ex) when (IsAuthException(ex)) { throw; }`. This empty catch-and-rethrow is needed to stop the generic `catch` from swallowing auth failures, but it looks accidental without context. Add a comment: `// Re-throw auth failures so the outer handler can audit already-renamed items before propagating.` |

File: backend/CPS.ComplexCases.API/CPS.ComplexCases.API.csproj [hunk 6]
```diff
@@ -29,7 +29,7 @@
     <PackageReference Include="Microsoft.Azure.Functions.Worker.ApplicationInsights" Version="2.0.0" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" Version="1.1.7" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore" Version="2.1.0" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="2.0.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="2.0.2" />
```
Comment (HasanCPS) [inline backend/CPS.ComplexCases.API/CPS.ComplexCases.API.csproj]: `Microsoft.Azure.Functions.Worker.Sdk` bump 2.0.0 to 2.0.2 is unrelated to the refactoring. Bundling dependency updates with behaviour-preserving refactors makes rollback harder if one causes issues. Move the package bump to its own commit or PR.

**Review by HasanCPS — APPROVED** [review]
LGTM!
